Consider removing future as a dependency

I use `zenpy` as dependency on a project, we have many months seeing a weird warning while running our tests due to `future` raising a `DeprecationWarning`, by using the now deprecated `imp` module.

Yeah, I know we can just ignore this specific warning, but that seems "dirty". Please note that `future` has some issues about this, the oldest one has more than 3 years open without any change...
https://github.com/PythonCharmers/python-future/issues/246

In the same issue above, there is a reference to a `pysaml2`'s pull request where the maintainer tells that he will use `six` instead of `future`.
https://github.com/IdentityPython/pysaml2/pull/576

Here's the traceback shown when I run our tests with `pytest -W error`:
```
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
django: settings: project.settings.test (from env)
rootdir: /opt/project/project, inifile: pytest.ini
plugins: xdist-1.31.0, factoryboy-2.0.3, cov-2.8.1, django-3.8.0, forked-1.1.3, mock-2.0.0
collecting 0 items / 1 error                                                                                                                                                                                        
====================================================================================================== ERRORS =======================================================================================================
__________________________________________________________________________________ ERROR collecting apps/contracts/api/v1/tests.py __________________________________________________________________________________
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
apps/contracts/api/v1/__init__.py:1: in <module>
    from .tasks import *  # NOQA
apps/contracts/api/v1/tasks.py:1: in <module>
    from zenpy import Zenpy
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/zenpy/__init__.py:6: in <module>
    from zenpy.lib.api import (
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/zenpy/lib/api.py:43: in <module>
    from zenpy.lib.response import *
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/zenpy/lib/response.py:4: in <module>
    from zenpy.lib.generator import SearchResultGenerator, ZendeskResultGenerator, ChatResultGenerator, ViewResultGenerator, \
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/zenpy/lib/generator.py:8: in <module>
    from future.standard_library import install_aliases
/home/python/.cache/pypoetry/virtualenvs/project-soWN0cI--py3.7/lib/python3.7/site-packages/future/standard_library/__init__.py:65: in <module>
    import imp
/usr/local/lib/python3.7/imp.py:33: in <module>
    DeprecationWarning, stacklevel=2)
E   DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================================= 1 error in 4.53s ==================================================================================================
```

Any way to have this fixed? :pray:

--- a/requirements.dev
+++ b/requirements.dev
@@ -1,7 +1,6 @@
 cachetools>=3.1.0
 requests>=2.14.2
 pytz>=2018.9
-future>=0.17.1
 python-dateutil>=2.7.5
 sphinx>=1.8.4
 pycodestyle>=2.5.0
@@ -13,3 +12,4 @@ yapf>=0.26.0
 bs4>=0.0.1
 lxml>=4.3.1
 sphinx_rtd_theme>=0.4.3
+six>=1.14.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 cachetools>=3.1.0
 requests>=2.14.2
 pytz>=2018.9
-future>=0.17.1
 python-dateutil>=2.7.5
+six>=1.14.0

--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ setup(
         'python-dateutil>=2.7.5',
         'cachetools>=3.1.0',
         'pytz>=2018.9',
-        'future>=0.17.1'
+        'six>=1.14.0',
     ],
     keywords=['zendesk', 'api', 'wrapper'],
     classifiers=[

--- a/zenpy/lib/generator.py
+++ b/zenpy/lib/generator.py
@@ -5,8 +5,6 @@ import re
 from abc import abstractmethod
 from datetime import datetime, timedelta
 
-from future.standard_library import install_aliases
-
 from zenpy.lib.util import as_plural
 from zenpy.lib.exception import SearchResponseLimitExceeded
 
@@ -15,7 +13,7 @@ try:
 except ImportError:
     from collections import Iterable
 
-install_aliases()
+import six
 from math import ceil
 
 __author__ = 'facetoe'
@@ -124,6 +122,10 @@ class BaseResultGenerator(Iterable):
         # Calculate our range of pages.
         min_page = ceil(start / page_size)
         max_page = ceil(stop / page_size) + 1
+
+        if six.PY2:
+            min_page = int(min_page)
+            max_page = int(max_page)
 
         # Calculate the lower and upper bounds for the final slice.
         padding = ((max_page - min_page) - 1) * page_size

--- a/zenpy/lib/response.py
+++ b/zenpy/lib/response.py
@@ -4,7 +4,7 @@ from zenpy.lib.exception import ZenpyException
 from zenpy.lib.generator import SearchResultGenerator, ZendeskResultGenerator, ChatResultGenerator, ViewResultGenerator, \
     TicketAuditGenerator, ChatIncrementalResultGenerator, JiraLinkGenerator
 from zenpy.lib.util import as_singular, as_plural, get_endpoint_path
-from urllib.parse import urlparse
+from six.moves.urllib.parse import urlparse
 
 
 class ResponseHandler(object):

